.github/jobs/configure-checks/all.bats

```diff
@@ -0,0 +1,371 @@
+#!/usr/bin/env bats
+
+load 'assert'
+
+u="domjudge-bats-user"
+
+setup_user() {
+    id -u $u || useradd $u >/dev/null
+    chown -R $u:$u ./
+}
+
+setup() {
+    setup_user
+    for shared_file in config.log confdefs.h conftest.err; do
+        chmod a+rw $shared_file || true
+    done
+    repo-install gcc g++ libcgroup-dev
+}
+
+q_configure () {
+    su $u -c "./configure $@"
+    #./configure $@ --with-domjudge-user=root
+}
+
+repo-install () {
+    apt-get install $@ -y >/dev/null
+}
+repo-remove () {
+    apt-get remove $@ -y >/dev/null; apt-get autoremove -y 2>/dev/null
+}
+remove_gnuc () {
+    repo-remove gcc g++
+}
+remove_c () {
+    remove_gnuc; repo-remove clang
+}
```

Comment:
Nitpick: I think there's a bit too much indirection here: why declare functions `remove_{gnuc,c}` that are used twice, but do directly call `repo-{remove,install}` below for other things? I'd just use the latter functions directly.

Reply:
Sure, when writing the tests I expected to need it more often, I agree at this moment it doesn't really help.